libs/src/service-selection/ServiceSelection.ts

```diff
@@ -248,47 +308,47 @@ class ServiceSelection {
   }
 
   /** Where does the health check for this type of service live */
-  static getHealthCheckEndpoint (service) {
+  static getHealthCheckEndpoint (service: string) {
     return `${service}/health_check`
   }
 
   /**
    * What the criteria is for a healthy service
-   * @param {Object} response axios response
+   * @param response axios response
    * @param {{ [key: string]: string}} urlMap health check urls mapped to their cannonical url
    * e.g. https://discoveryprovider.audius.co/health_check => https://discoveryprovider.audius.co
    */
-  isHealthy (response, urlMap) {
+  isHealthy (response: AxiosResponse, _urlMap: Record<string, string>) {
     return response.status === 200
   }
 
   /** Races requests against each other with provided timeouts and health checks */
-  async race (services) {
+  async race (services: string[]) {
     // Key the services by their health check endpoint
-    const map = services.reduce((acc, s) => {
+    const serviceMap = services.reduce<Record<string, string>>((acc, s) => {
       acc[ServiceSelection.getHealthCheckEndpoint(s)] = s
       return acc
     }, {})
 
-    let best
+    let best: string | null | undefined = null
```

Comment:
yeah good q, it was undefined because 
<img width="567" alt="image" src="https://user-images.githubusercontent.com/8230000/160198651-aa899d2b-146a-444b-8889-6aa28530fd7e.png">

Thats cause typescript can't guarantee `serviceMap[url]` returns string, cause serviceMap might not have url in it. however maybe we can make the assumption that it always does, so ill update it to `serviceMap[url] as string`. nice catch

Reply:
thanks!